containers/src/Kokkos_DynamicView.hpp

```diff
@@ -819,9 +819,9 @@ inline void deep_copy(const ExecutionSpace& exec,
 
   // FIXME use execution space
   if (DstExecCanAccessSrc)
-    Kokkos::Impl::ViewRemap<dst_type, src_type, dst_execution_space>(dst, src);
+    Kokkos::Impl::ViewRemap<dst_type, src_type>(dst, src);
   else if (SrcExecCanAccessDst)
-    Kokkos::Impl::ViewRemap<dst_type, src_type, src_execution_space>(dst, src);
+    Kokkos::Impl::ViewRemap<dst_type, src_type>(dst, src);
```

Comment:
Consider doing `if (DstExecCanAccessSrc || SrcExecCanAccessDst)` since the two branches are now identical